guide/additional-info/changes-in-v13.md

```diff
@@ -0,0 +1,384 @@
+
+## Commonly used methods that changed
+
+### Intents
+
+As v13 makes the switch to Discord API v8, it will now be **required** to specify intents in your Client constructor.
+For more information, refer to our more [detailed article about this topic](/popular-topics/intents).
+
+:::danger
+
+`clientOptions.disableMentions` has been removed and replaced with `clientOptions.allowedMentions`!
+
+The Discord API now allows bots much more granular control over mention parsing, down to the specific ID if desired. Refer to the [Discord API documentation](https://discord.com/developers/docs/resources/channel#allowed-mentions-object) for more information.
+
+```diff
+- const client = new Discord.Client({ disableMentions: 'everyone' });
++ const client = new Discord.Client({ allowedMentions: { parse: ['users', 'roles'], repliedUser: true } });
+```
+
+:::
+
+### Replies / Message#reply
+
+`Message.reply()` will no longer result in the bot prepending a user mention to the content, replacing the behaviour with Discord's reply feature.
+
+`MessageOptions.reply` (User ID) has been removed, replaced with `MessageOptions.replyTo` (Message ID)
+
+```diff
+- channel.send('content', { reply: '123456789012345678' })` // User ID
++ channel.send('content', { replyTo: '765432109876543219' })` // Message ID
+```
+
+The new `MessageOptions.allowedMentions.repliedUser` boolean option determines if the reply will notify the author of the original message.
+
+```diff
+- message.reply('content')
++ message.reply('content', { allowedMentions: { repliedUser: false }})
+```
+
+### Bitfields / Permissions
+
+The usage of string literals for bitfields such as `Permissions` and `UserFlags` is discouraged, you should use the flag instead.
+
+```diff
+- permissions.has('SEND_MESSAGES')
++ permissions.has(Permissions.FLAGS.SEND_MESSAGES)
+```
+
+### Webpack
+
+Webpack builds are no longer supported.
+
+## Changes and deletions
+
+### Client
+
+#### Client#emojis
+
+The Client Emoji manager is now a `BaseGuildEmojiManager`, providing cache resolution only and removing methods which would fail to create emojis as there was no Guild context.
+
+#### Client#generateInvite
+
+`Client#generateInvite` no longer supports `PermissionsResolvable` as its argument, requiring `InviteGenerationOptions`.
+
+To provide permissions, use `InviteGenerationOptions#permissions`.
+```diff
+- client.generateInvite([Permissions.FLAGS.SEND_MESSAGES]);
++ client.generateInvite({ permissions: [Permissions.FLAGS.SEND_MESSAGES] })
+```
+
+### ClientOptions
+
+#### ClientOptions#fetchAllMembers
+
+The `ClientOptions#fetchAllMembers` option has been removed.
+
+With the introduction of gateway intents, the `fetchAllMembers` Client option would often fail and causes major delays in ready states or even cause timeout errors. As its purpose is contradictory to Discord's intentions to reduce scraping of user and presence data, it has been removed.
```

Comment:
will there be an example code for an alternative here or elsewhere in the guide? while this shouldn't be recommended, it is unfortunately needed for getting previous state in events (and unfortunately d.js defaults to dropping the event if it isn't cached)

Reply:
There is no alternative to it directly.

The only alternative and always recommended way was fetching members when needed, or just calling `.fetch()` directly on the guilds members collection if you really need it.

Reply:
https://github.com/discordjs/discord.js/blob/ee5bc1a5c4e218ea1f16992694a2bff86c86d2d2/test/random.js#L21-L40 was what I added when we removed the opt, which mostly achieves the same thing, not perfect though.